tests/rules/test_quoted_strings.py

```diff
@@ -35,102 +71,59 @@ def test_disabled(self):
-                   'string2: "foo"\n'
-                   'string3: \'true\'\n'                     # fails
-                   'string4: \'123\'\n'                      # fails
-                   'string5: \'bar\'\n'                      # fails
-                   'string6: !!str genericstring\n'
-                   'string7: !!str 456\n'
-                   'string8: !!str "quotedgenericstring"\n'
-                   'binary: !!binary binstring\n'
-                   'integer: !!int intstring\n'
-                   'boolean2: !!bool boolstring\n'
-                   'boolean3: !!bool "quotedboolstring"\n',
-                   conf, problem1=(4, 10), problem2=(6, 10),
+
+        self.check(YAML, conf, problem1=(4, 10), problem2=(8, 10))
+        self.check(YAML_MULTILINE, conf, problem1=(9, 3))
+
+    def test_disallow_redundant_quotes(self):
+        conf = 'quoted-strings: {required: only-when-needed}\n'
+
+        self.check(YAML, conf, problem1=(5, 10), problem2=(8, 10))
+        self.check(YAML_MULTILINE, conf, problem1=(12, 3))
+
+    def test_disallow_redundant_single_quotes(self):
+        conf = 'quoted-strings: {quote-type: single, ' + \
+                                'required: only-when-needed}\n'
+
+        self.check(YAML, conf, problem1=(5, 10), problem2=(6, 10),
                    problem3=(7, 10), problem4=(8, 10))
-        self.check('---\n'
-                   'multiline string 1: |\n'
-                   '  line 1\n'
-                   '  line 2\n'
-                   'multiline string 2: >\n'
-                   '  word 1\n'
-                   '  word 2\n'
-                   'multiline string 3:\n'
-                   '  word 1\n'
-                   '  word 2\n'
-                   'multiline string 4:\n'
-                   '  "word 1\\\n'
-                   '   word 2"\n',
-                   conf, problem1=(9, 3))
+        self.check(YAML_MULTILINE, conf, problem1=(12, 3))
+
+    def test_single_quotes_required(self):
+        conf = 'quoted-strings: {quote-type: single, required: true}\n'
+
+        self.check(YAML, conf, problem1=(4, 10), problem2=(5, 10),
+                   problem3=(6, 10), problem4=(7, 10))
+        self.check(YAML_MULTILINE, conf, problem1=(9, 3), problem2=(12, 3))
+
+    def test_any_quotes_relaxed(self):
+        conf = 'quoted-strings: {quote-type: any, required: false}\n'
+
+        self.check(YAML, conf)
+        self.check(YAML_MULTILINE, conf)
+
+    def test_single_quotes_relaxed(self):
+        conf = 'quoted-strings: {quote-type: single, required: false}\n'
+
+        self.check(YAML, conf, problem2=(5, 10),
+                   problem3=(6, 10), problem4=(7, 10))
+        self.check(YAML_MULTILINE, conf, problem1=(12, 3))
+
+    def test_quotes_required(self):
+        conf = 'quoted-strings: {quote-type: any, required: true}\n'
+
+        self.check(YAML, conf, problem2=(4, 10))
+        self.check(YAML_MULTILINE, conf, problem1=(9, 3))
```

Comment:
I would also have expected lines 10 and 11 to be reported:

```yaml
string6: !!str genericstring
string7: !!str 456
```
... but it's not currently the case on master with `{quote-type: any}`. I think that's acceptable!

Reply:
I didn't change the previous behavior and I actually agree with it: if, for some reason, you are explicitly defining the type, you should know what you're doing. Makes sense?

Reply:
Agreed!